streams/src/test/java/org/apache/kafka/streams/tests/EosTestDriver.java

```diff
@@ -144,6 +188,14 @@ public static void verify(final String kafka, final boolean withRepartitioning)
         props.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, ByteArrayDeserializer.class);
         props.put(ConsumerConfig.ISOLATION_LEVEL_CONFIG, IsolationLevel.READ_COMMITTED.toString().toLowerCase(Locale.ROOT));
 
+        try (final KafkaConsumer<byte[], byte[]> committedConsumer = new KafkaConsumer<>(props)) {
+            verifyAllTransactionFinished(committedConsumer, kafka, withRepartitioning);
```

Comment:
To make the verification step work, we _first_ need to check that all transactions are finished. With EOS-alpha we never had pending transactions that would eventually be aborted by the tx-coordinator, because while we crash some instanced in between the final shutdown phase is always clean. Hence, for eos-alpha all pending transactions would be aborted by `initTransaction()` calls.

For eos-beta, thread that are killed leave open transaction that will be eventually expired by the tx-coordinator though, as we (also on restart of a thread) would generate a new `transactonal.id`.

Having no pending transactions is a requirement for the following code to do a correct verification of the result.